### PR TITLE
[Refactor] Networking package Part 1 #391

### LINKED ISSUE #391: Refactor NetworkInteractor for usage in all APIs

### PR DESCRIPTION
**Pull request by flypaper0**
# Description

- WalletConnectNetwork package
- package connected to Auth and Chat SDK

Resolves #391

## How Has This Been Tested?
Run unit/integration tests

## Due Dilligence

* [ ] Breaking change
* [ ] Requires a documentation update

### CODE REVIEW

File: Sources/Chat/ProtocolServices/Invitee/InvitationHandlingService.swift [hunk 1]
```diff
@@ -73,47 +70,36 @@ class InvitationHandlingService {
...
-                } catch {
-                    logger.debug("Did not handle invite, error: \(error)")
-                }
-            default:
-                return
-            }
-        }.store(in: &publishers)
-    }
+        networkingInteractor.requestPublisher.sink { [unowned self] payload in
+            guard payload.request.method == "wc_chatInvite"
```
Comment (llbartekll) [inline Sources/Chat/ProtocolServices/Invitee/InvitationHandlingService.swift:87]: what do you think if at some point we have an enum with protocol methods? so we do like:
```guard payload.request.method == ChatProtocolMethods.invite.rawValue```

File: Sources/Chat/Types/Message.swift [hunk 2]
```diff
@@ -20,4 +13,19 @@ public struct Message: Codable, Equatable {
         case authorAccount
         case timestamp
     }
+
+    static var tag: Int {
+        return 2002
+    }
+
+    static var method: String {
+        return "wc_chatMessage"
```
Comment (llbartekll) [inline Sources/Chat/Types/Message.swift:22]: this could be a ChatProtocolMethods case as well

**Review by llbartekll — APPROVED** [review]

File: Sources/Chat/ProtocolServices/Invitee/InvitationHandlingService.swift (hunk 1 above)
Comment (flypaper0) [reply Sources/Chat/ProtocolServices/Invitee/InvitationHandlingService.swift:87]: I will solve these issue in next PR

File: Sources/Chat/ChatClientFactory.swift [hunk 3]
```diff
@@ -10,17 +11,18 @@ public struct ChatClientFactory {
...
         kms: KeyManagementService,
         logger: ConsoleLogging,
-        keyValueStorage: KeyValueStorage) -> ChatClient {
+        keyValueStorage: KeyValueStorage
+    ) -> ChatClient {
         let topicToRegistryRecordStore = CodableStore<RegistryRecord>(defaults: keyValueStorage, identifier: StorageDomainIdentifiers.topicToInvitationPubKey.rawValue)
         let serialiser = Serializer(kms: kms)
-        let jsonRpcHistory = JsonRpcHistory<ChatRequestParams>(logger: logger, keyValueStore: CodableStore<JsonRpcRecord>(defaults: keyValueStorage, identifier: StorageDomainIdentifiers.jsonRpcHistory.rawValue))
-        let networkingInteractor = NetworkingInteractor(relayClient: relayClient, serializer: serialiser, logger: logger, jsonRpcHistory: jsonRpcHistory)
+        let rpcHistory = RPCHistory(keyValueStore: CodableStore<RPCHistory.Record>(defaults: keyValueStorage, identifier: StorageDomainIdentifiers.jsonRpcHistory.rawValue))
```
[the remark this answers is not in the document]
Comment (flypaper0) [reply Sources/Chat/ChatClientFactory.swift:18]: decided to not make migration

**Pull request merged**